platform/engineering/request-for-change/2020-01-08-ssoe-removing-implicit-mvi-queries.md

```diff
@@ -0,0 +1,106 @@
+# RFC: Removing implicit MVI lookups
+
+- Date: _2020-01-08_
+
+
+## Background
+At present, when a user authenticates and is determined to be Level of assurance 3, we look up that user's VA profile in MVI (aka MPI, used interchangeably hereafter). The combination of the incoming attributes from the identity provider and the attributes from MVI make up the bulk of the logical user model that is used across vets-api.
+
+As part of the overall SSO consolidation work, we are integrating VA SSOe with the VA.gov sign-in process. We will maintain the three current credential options, but the SAML request/response cycle will be intercepted by SSOe before being returned to VA.gov. Along the way, SSOe can establish a session cookie for use by other sites. But more importantly, SSOe will provide a standardized SAML payload that includes attributes from the credential provider AND from MVI. 
```

Comment:
question from someone who's not close to the rest of the SSOe work: how do different services (vets-api, ...[other services using SSOe]) all get access to the SAML payload, if it's just identified by a session cookie? 

another way to ask: where is that SAML payload cached/stored? it sounds like it's used by a broader set of services than just vets-api

Reply:
SSOe has all the content for the SAML payload stored in association with the user's SSOe session. So another system would make their own SAML request to SSOe and SSOe would send them a SAML response. Because a session already exists (i.e. SSOe can detect their cookie and use the token inside to look up the user's session), SSOe does not need to trigger user authentication - so there's no user interaction other than seeing their browser redirect a couple times.